lib/improver/tests/weighted_blend/test_blending_utilities.py

```diff
@@ -0,0 +1,230 @@
+                                    "time", units=tunit), 0)
+        self.cube = cube
+        new_scalar_coord = iris.coords.AuxCoord(1,
+                                                long_name='dummy_scalar_coord',
+                                                units='no_unit')
+        cube_with_scalar = cube.copy()
+        cube_with_scalar.add_aux_coord(new_scalar_coord)
+        self.cube_with_scalar = cube_with_scalar
+
+    def test_basic_weighted_average(self):
+        """Test basic_weighted_average works."""
+        coord = "time"
+        weights = [0.8, 0.2]
+        coord_dim = self.cube.coord_dims(coord)
+        result = BlendingUtilities.basic_weighted_average(self.cube,
+                                                          weights,
+                                                          coord,
+                                                          coord_dim)
+        expected_result_array = np.ones((2, 2))*1.2
+        self.assertArrayAlmostEqual(result.data, expected_result_array)
+
+    def test_blend_percentile_cube1(self):
+        """Test blending percentile cube works with equal weights."""
+        coord = "time"
+        weights = np.array([0.5, 0.5])
+        perc_cube = percentile_cube()
+        coord_dim = perc_cube.coord_dims(coord)
+        perc_coord = perc_cube.coord('percentile_over_realization')
+        result = BlendingUtilities.blend_percentile_cube(perc_cube,
+                                                         weights,
+                                                         coord,
+                                                         coord_dim,
+                                                         perc_coord)
+        expected_result_array = np.reshape(BLENDED_PERCENTILE_DATA1,
+                                           (6, 2, 2))
+        self.assertArrayAlmostEqual(result.data, expected_result_array)
+
+    def test_blend_percentile_cube2(self):
+        """Test blending percentile cube works with unequal weights."""
+        coord = "time"
+        weights = np.array([0.8, 0.2])
+        perc_cube = percentile_cube()
+        coord_dim = perc_cube.coord_dims(coord)
+        perc_coord = perc_cube.coord('percentile_over_realization')
+        result = BlendingUtilities.blend_percentile_cube(perc_cube,
+                                                         weights,
+                                                         coord,
+                                                         coord_dim,
+                                                         perc_coord)
+        expected_result_array = np.reshape(BLENDED_PERCENTILE_DATA2,
+                                           (6, 2, 2))
+        self.assertArrayAlmostEqual(result.data, expected_result_array)
+
+    def test_blend_percentile_aggregate(self):
+        """Test blend_percentile_aggregate function works"""
+        weights = np.array([0.8, 0.2])
+        percentiles = np.array([0, 20, 40, 60, 80, 100])
```

Comment:
As discussed, the plugin only accepts a cube but there is a more general task to reduce incoming data onto a cube. #209 has been raised for this work.